Tests/UnitTests/Core/Utilities/GridIterationTests.cpp

```diff
@@ -0,0 +1,373 @@
+                          Acts::detail::EquidistantAxis,
+                          Acts::detail::EquidistantAxis>
+      gridStop = grid.end(navigation);
+  std::size_t numIterations = 0ul;
+  for (; gridStart != gridStop; ++gridStart) {
+    ++numIterations;
+  }
+  BOOST_CHECK_EQUAL(numIterations, grid.size(false));
+}
+
+BOOST_AUTO_TEST_CASE(grid_iteration_test_5d_local_custom_subnavigation) {
+  const std::size_t nBins = 10ul;
+  const std::size_t nBinsZ = 20ul;
+  const std::size_t nBinsJK = 5ul;
+  Acts::detail::EquidistantAxis xAxis(0, 100, nBins);
+  Acts::detail::EquidistantAxis yAxis(0, 100, nBins);
+  Acts::detail::EquidistantAxis zAxis(0, 100, nBinsZ);
+  Acts::detail::EquidistantAxis jAxis(0, 100, nBinsJK);
+  Acts::detail::EquidistantAxis kAxis(0, 100, nBinsJK);
+  Acts::Grid<double, Acts::detail::EquidistantAxis,
+             Acts::detail::EquidistantAxis, Acts::detail::EquidistantAxis,
+             Acts::detail::EquidistantAxis, Acts::detail::EquidistantAxis>
+      grid(std::make_tuple(std::move(xAxis), std::move(yAxis), std::move(zAxis),
+                           std::move(jAxis), std::move(kAxis)));
+
+  // test general properties
+  BOOST_CHECK_EQUAL(grid.size(false),
+                    nBins * nBins * nBinsZ * nBinsJK * nBinsJK);
+  BOOST_CHECK_EQUAL(grid.size(true), (nBins + 2ul) * (nBins + 2ul) *
+                                         (nBinsZ + 2ul) * (nBinsJK + 2ul) *
+                                         (nBinsJK + 2ul));
+
+  const std::array<std::size_t, 5ul> numLocalBins = grid.numLocalBins();
+  BOOST_CHECK_EQUAL(numLocalBins[0ul], nBins);
+  BOOST_CHECK_EQUAL(numLocalBins[1ul], nBins);
+  BOOST_CHECK_EQUAL(numLocalBins[2ul], nBinsZ);
+  BOOST_CHECK_EQUAL(numLocalBins[3ul], nBinsJK);
+  BOOST_CHECK_EQUAL(numLocalBins[4ul], nBinsJK);
+
+  // Iterate only on a few bins
+  std::array<std::vector<std::size_t>, 5ul> navigation;
+  navigation[0ul] = {1ul, 5ul, 3ul, 2ul, 9ul, 10ul, 4ul, 6ul, 8ul, 7ul};
+  navigation[1ul] = {6ul, 8ul, 7ul, 1ul};
+  navigation[2ul] = {1ul, 5ul};
+  navigation[3ul] = {5ul, 3ul, 2ul};
+  navigation[4ul] = {2ul};
+
+  Acts::GridLocalIterator<
+      double, Acts::detail::EquidistantAxis, Acts::detail::EquidistantAxis,
+      Acts::detail::EquidistantAxis, Acts::detail::EquidistantAxis,
+      Acts::detail::EquidistantAxis>
+      gridStart = grid.begin(navigation);
+  Acts::GridLocalIterator<
+      double, Acts::detail::EquidistantAxis, Acts::detail::EquidistantAxis,
+      Acts::detail::EquidistantAxis, Acts::detail::EquidistantAxis,
+      Acts::detail::EquidistantAxis>
+      gridStop = grid.end(navigation);
+  std::size_t numIterations = 0ul;
+  for (; gridStart != gridStop; ++gridStart) {
+    ++numIterations;
+  }
+
+  std::size_t expectedIterations = 1ul;
+  for (std::size_t i(0ul); i < 5ul; ++i) {
+    expectedIterations *= navigation[i].size();
+  }
+
+  BOOST_CHECK_EQUAL(numIterations, expectedIterations);
+}
+
```

Comment:
Is there also a test that accesses values and checks that the iterator hits all of the right positions? Or is it enough that the seeding tests pass?

Reply:
I can add something of this kind indeed

Reply:
I have added a test for it